extensions/guacamole-auth-restrict/src/main/java/org/apache/guacamole/auth/restrict/RestrictionVerificationService.java

```diff
@@ -0,0 +1,337 @@
+            if (!allowedByHostRestrictions(allowedHostString, deniedHostString, remoteAddress))
+                throw new TranslatableInvalidHostLoginException("User \"" 
+                        + authenticatedUser.getIdentifier() 
+                        +"\" is not allowed to log in from \"" 
+                        + remoteAddress + "\"",
+                    "RESTRICT.ERROR_USER_LOGIN_NOT_ALLOWED_FROM_HOST");
+        
+        // Gather user's effective groups.
+        Set<String> userGroups = authenticatedUser.getEffectiveUserGroups();
+        Directory<UserGroup> directoryGroups = context.getPrivileged().getUserGroupDirectory();
+        
+        // Loop user's effective groups and verify restrictions
+        for (String userGroup : userGroups) {
+            UserGroup thisGroup = directoryGroups.get(userGroup);
+            if (thisGroup == null) {
+                continue;
+            }
+
+            // Get group's attributes
+            Map<String, String> grpAttributes = thisGroup.getAttributes();
+            
+            // Pull time-based restrictions for this group and verify
+            String grpAllowedTimeString = grpAttributes.get(RestrictUserGroup.RESTRICT_TIME_ALLOWED_ATTRIBUTE_NAME);
+            String grpDeniedTimeString = grpAttributes.get(RestrictUserGroup.RESTRICT_TIME_DENIED_ATTRIBUTE_NAME);
+            if ((grpAllowedTimeString != null && !grpAllowedTimeString.isEmpty()) || (grpDeniedTimeString != null && !grpDeniedTimeString.isEmpty()))
+                if (!allowedByTimeRestrictions(grpAllowedTimeString, grpDeniedTimeString))
+                    throw new TranslatableInvalidTimeLoginException("User \"" 
+                            + authenticatedUser.getIdentifier() 
+                            +"\" is not allowed to log in at this time due to restrictions on group \"" 
+                            + userGroup + "\".",
+                            "RESTRICT.ERROR_USER_LOGIN_NOT_ALLOWED_NOW");
+            
+            // Pull host-based restrictions for this group and verify
+            String grpAllowedHostString = grpAttributes.get(RestrictUserGroup.RESTRICT_HOSTS_ALLOWED_ATTRIBUTE_NAME);
+            String grpDeniedHostString = grpAttributes.get(RestrictUserGroup.RESTRICT_HOSTS_DENIED_ATTRIBUTE_NAME);
+            if ((grpAllowedHostString != null && !grpAllowedHostString.isEmpty()) || (grpDeniedHostString != null && !grpDeniedHostString.isEmpty()))
+                if (!allowedByHostRestrictions(grpAllowedHostString, grpDeniedHostString, remoteAddress))
+                    throw new TranslatableInvalidHostLoginException("User \"" 
+                            + authenticatedUser.getIdentifier() 
+                            + "\" is not allowed to log in from this host due to restrictions on group \"" 
+                            + userGroup + "\".",
+                        "RESTRICT.ERROR_USER_LOGIN_NOT_ALLOWED_FROM_HOST");
```

Comment:
This looks to be mis-indented.

Reply:
Ugh, yes. The new version of NetBeans has a "feature" enabled that puts extra stuff at the beginning of the parameters, and it's messing up my ability to indent correctly...